### PR TITLE
Fix #1508: Featured challenge on landing page

### LINKED ISSUE #1508: Featured challenge on home page
## Deliverables

- [ ] On the home page, the featured challenge is not selected on the basis of featured flag, but on the basis of approved_by_admin and published flag.

### PR DESCRIPTION
**Pull request by viditjain08**
- Landing page now only showing the featured challenge if it has featured=True, approved_by_admin=True, published=True and active

### CODE REVIEW

**Comment by viditjain08** [comment]
@deshraj, @aka-jain, @RishabhJain2018, please review this.

**Comment by aka-jain** [comment]
@deshrah I think for featured challenge we had decided to go with the present challenges and making it featured by admin flag.

Correct if I'm wrong or if we need different calls for featured challenges.

**Comment by deshraj** [comment]
Yeah, you are correct. This is what @viditjain08 has done actually. Thanks @viditjain08 for adding this. I will review this pull request in sometime. Assigning to myself.

File: apps/challenges/views.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def get_all_challenges(request, challenge_time):
     Returns the list of all challenges
     """
     # make sure that a valid url is requested.
-    if challenge_time.lower() not in ("all", "future", "past", "present"):
+    if challenge_time.lower() not in ("all", "future", "past", "present", "featured"):
```
Comment (deshraj) [inline apps/challenges/views.py]: It's a bad practice to add `featured` as one of the value of `challenge_time` attribute. Instead please use `featured` as a different query parameter in this endpoint.

**Comment by deshraj** [comment]
@viditjain08: any update on this?

**Comment by deshraj** [comment]
@viditjain08: can you please let us know that when you will be able to resolve the comments? Please let us know as well if you won't be able to work on this. Someone else can take it forward from here. :)

**Comment by deshraj** [comment]
Closing due to inactivity.

**Pull request closed without merging**